lambdas/service/app.py

```diff
@@ -650,6 +652,19 @@ def validate_filters(filters):
             else:
                 raise BadRequestError(f'The relation in the `filters` parameter entry for `{facet}`'
                                       f' must be one of {valid_relations}')
+            if facet == 'organismAge':
+                for f in value:
+                    try:
+                        value, unit = f['value'], f['unit']
+                    except KeyError:
+                        raise BadRequestError(
+                            msg=f'The `{facet}` filter parameter should specify a dictionary with `value` and `unit` '
+                                f'keys. For filtering on the absence of this fields use `None`.')
+                    reject('' in (value, unit),
+                           f'Facet {facet} should not specify empty strings values',
+                           exception=BadRequestError)
+                    if value is None:
+                        require(unit is None, 'A unit without a value is not valid.', exception=BadRequestError)
```

Comment:
Setting a variable named `value` (line 658) while inside a loop of a variable with the same name `for f in value` is probably not a good idea.
Instead of creating the variables `unit` and `value` how about something like this?
```
            if facet == 'organismAge':
                for f in value:
                    try:
                        reject('' in (f['value'], f['unit']),
                               f'Facet {facet} should not specify empty strings values',
                               exception=BadRequestError)
                    except KeyError:
                        raise BadRequestError(
                            msg=f'The `{facet}` filter parameter should specify a dictionary with `value` and `unit` '
                                f'keys. For filtering on the absence of this fields use `None`.')
                    if f['value'] is None:
                        require(f['unit'] is None, 'A unit without a value is not valid.', exception=BadRequestError)
```